### PR TITLE
closes #21

### LINKED ISSUE #21: Don't use env variables for passing variables to bash script
The commands in the filter json files are just lines of bash shell script. Right now we pass the variables from the interface to this script by defining them as environment variables when the line of shell script is executed. But they are also inherited by all the subprocesses at that point.

Now say you define a placeholder/variable `LANG` in your filter script. And that script will call some perl script to do the actual filtering. Now Perl will complain that the LANG environment variable is not set to a valid language.

Ideally we can somehow pass variables to the bash script in a way that it is still Bash that does the variable expansion (because Bash has quite a few nifty features when doing that) but not cause the variables to be inherited by the subprocesses.

### PR DESCRIPTION
**Pull request by ales-t**

### CODE REVIEW

**Comment by jelmervdl** [comment]
It looks like it is breaking when used with monolingual filters (e.g. _deescape-special-chars_). Probably because that also wraps the command in a `col.py` call.

Error:
```
/bin/sh: -c: line 0: syntax error near unexpected token `;'
/bin/sh: -c: line 0: `; /Users/jelmer/Workspace/statmt/empty-train/col.py 1 ./deescape-special-chars.perl'
```

**Comment by ales-t** [comment]
@jelmervdl It's not about the monolingual filters specifically, I think. It's just that the filter doesn't specify any variables which is an edge case I didn't handle. You can check the updated version.

File: main.py [hunk 1]
```diff
@@ -276,11 +280,19 @@ async def get_sample(name:str, filters:List[FilterStep]) -> AsyncIterator[Filter
...
 
+        params = {name: props.export(filter_step.parameters[name])
+                  for name, props in filter_definition.parameters.items()}
+
+        if params:
+            vars_setter = '; '.join(f"{k}={quote(v)}" for k, v in params.items())
+            command = f'{vars_setter}; {command}'
+
+        print(command)
+
```
Comment (jelmervdl) [inline main.py:291]:
```suggestion
```

**Review by jelmervdl — CHANGES_REQUESTED** [review]
Looks good to me! Just need to remove the left-over `print(command)` debug code :)

**Review by jelmervdl — APPROVED** [review]

**Pull request merged**